components/server/ee/src/billing/billing-service.ts

```diff
@@ -43,8 +40,13 @@ export class BillingService {
             };
         }
 
-        const upcomingInvoice = await this.billingService.getUpcomingInvoice(attributionId);
-        const currentInvoiceCredits = upcomingInvoice.credits | 0;
+        const now = new Date();
+        const currentBalance = await this.usageService.listUsage({
```

Comment:
We could also specify Page Size 1 or 0 to avoid getting any results in the list response, we don't need it

Reply:
I use `from: now, to: now` which does also make sure we don't fetch usage entries

Reply:
Fair. We could change the usage implementation that if page size is 0, we don't even make the (list) query in the first place which would then be a bit faster.

Reply:
I'd rather introduce a new function theres too much information in the response that wouldn't make sense (e.g. balanceAtStart vs balanceAtEnd)

Reply:
your proposal makes sense once we need balance for a period but are not interested in the entries.